process/block/preprocess/sovereignChainTransactions.go

```diff
@@ -204,19 +198,39 @@ func (sct *sovereignChainTransactions) isTransactionEligibleForExecution(tx *tra
 	}
 
 	txFee := sct.economicsFee.ComputeTxFee(tx)
-	if accntInfo.balance.Cmp(txFee) < 0 {
-		log.Trace("sovereignChainTransactions.isTransactionEligibleForExecution", "error", process.ErrInsufficientFee,
-			"account balance", accntInfo.balance.String(),
-			"fee needed", txFee.String())
-		return process.ErrInsufficientFee, false
+	var cost *big.Int
+
+	if common.IsRelayedTxV3(tx) {
+		relayerAccount, relayerAccntInfo, errGetRelayer := sct.getRelayerAccountInfo(tx)
```

Comment:
on mx-chain-go we tried to avoid loading the same account twice, when when relayer is sender/receiver and it seems to be the same case here with `accountInfo` structure.. if they are the same, on L218 you you call setAccountInfo with the balance after fee.. then on L240 you setAccountInfo again, with balance - cost 

if relayer == sender || relayer == receiver, we simply return the pre-loaded accounts

this way we avoid invalid behavior while saving twice the same account

you can check `TestRelayedV3WithChainSimulator` from `integrationTests/chainSimulator/relayedTx_test.go` with some test examples (L59, L60)

Reply:
+1

Reply:
Fixed it.
Also made the setup to run all relayed tests also with sovereign chain simulator.